db/version_set.cc

```diff
@@ -1198,15 +1423,29 @@ bool LevelIterator::SkipEmptyFileForward() {
     if (file_index_ >= flevel_->num_files - 1) {
       // Already at the last file
       SetFileIterator(nullptr);
+      ClearRangeTombstoneIter();
       break;
     }
     if (KeyReachedUpperBound(file_smallest_key(file_index_ + 1))) {
       SetFileIterator(nullptr);
+      ClearRangeTombstoneIter();
       break;
     }
+    // may init a new *range_tombstone_iter
     InitFileIterator(file_index_ + 1);
+    // We moved to a new SST file
+    // Seek range_tombstone_iter_ to reset its !Valid() default state.
```

Comment:
Why is it needed here and SkipEmptyFileBackward(), but not in LevelIterator::Seek*() functions that may return a sentinel after InitFileIterator()?

Reply:
Good point, I was relying on an implicit assumption that when `LevelIterator::Seek*()` is called in merging iterator, that `Seek*` call will be accompanied by a similar `Seek*` into the corresponding range tombstone iterator, so there is no need to do the initial range tombstone iterator seek in `LevelIterator::Seek*()`. We could do the same initialization in LevelIterator too, that should not affect correctness.